### PR TITLE
Add OWASP Check on CI

### PR DESCRIPTION
**Pull request by JAVGan**
Add the step `OWASP check` into the `security` CI Job.

### CODE REVIEW

File: .github/workflows/tox-test.yml [hunk 1]
```diff
@@ -100,6 +100,64 @@ jobs:
...
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: owasp-report
+          path: |
+            dependency-check/dependency-check-report.html
+            dependency-check/dependency-check-report.xml
+            dependency-check/dependency-check-report.json
+            dependency-check/dependency-check-report.csv
+            dependency-check/dependency-check-report.sarif
```
Comment (lslebodn) [inline .github/workflows/tox-test.yml:160]: Is it expected there are not any artifacts?

image

File: .github/workflows/tox-test.yml [hunk 2]
```diff
@@ -100,6 +100,64 @@ jobs:
         uses: pypa/gh-action-pip-audit@v1.1.0
         with:
           inputs: requirements.txt requirements-test.txt
+      - name: Cache OWASP
+        uses: actions/cache@v3
```
Comment (lslebodn) [inline .github/workflows/tox-test.yml:104]: Could we try to use v4?

There is a warning
```
The following actions use a deprecated Node.js version and will be forced to run on node20: actions/checkout@v2, actions/setup-python@v2, actions/cache@v3. For more info: https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/
```
*  https://github.com/release-engineering/cloudpub/actions/runs/10515093079?pr=38

File: .github/workflows/tox-test.yml (hunk 1 above)
Comment (lslebodn) [reply .github/workflows/tox-test.yml:160]: nvm; I can see it on the another page
* https://github.com/release-engineering/cloudpub/actions/runs/10515093079?pr=38
image

Comment (JAVGan) [reply .github/workflows/tox-test.yml:160]: Yeah, it also shows the link within the execution logs

File: .github/workflows/tox-test.yml (hunk 2 above)
Comment (JAVGan) [reply .github/workflows/tox-test.yml:104]: Changed to v4 now, thanks for letting me know about the deprecation!

**Review by lslebodn — APPROVED** [review]
LGTM

**Pull request merged**